### DISCUSSION TITLE
Create a IDL with an array as a property of a struct

### DISCUSSION THREAD

**Discussion by AlbertoElDevaluador** [jmmorato/openddsharp#192] [Q&A] [answered]
How can i create a struct like that:

`struct ExampleStruct{
     long array[10];
}`
When i try to create executing the command: dotnet build ExampleStruct.csproj --runtime win-x64 --configuration Debug
Return an error and dont compile. 

And when i try to create a struct like that:
`struct ExampleStruct{
     long port;
}`
Return other error, its looks like the word port cant be used to create a variable

**Comment by jwillemsen** [comment]
`port` is a reserved keyword in IDL (see https://www.omg.org/spec/IDL/4.2/PDF chapter 7.4.2), you must add a `_` as prefix, for example see https://github.com/RemedyIT/taox11/blob/d1a33c8563529f907e5ea8a20d04b4ee92dbb0de/tests/keyword/test.idl#L50

**Reply by jwillemsen** [reply, marked as answer]
Also you must end the struct definition with a semi colon. It could be that an anonymous array is not supported (don't know all the details of what openddsharp supports), you could try using a typedef as we do here https://github.com/RemedyIT/taox11/blob/d1a33c8563529f907e5ea8a20d04b4ee92dbb0de/tests/array/test.idl#L24

**Reply by AlbertoElDevaluador** [reply]
And how can i create a class (struct) who inherit of other like this:
`struct ClassSon : ClassFather{
}
`

**Reply by jwillemsen** [reply]
struct inheritance is a IDL4 feature which isn't supported (yet) by OpenDDS and I assume also not openddsharp. You could create a `ClassSon` which contains a `ClassFather` as member as alternative.

**Reply by jmmorato** [reply]
Indeed: https://github.com/OpenDDS/OpenDDS/issues/4143

If not supported by OpenDDS not supported by OpenDDSharp

**Reply by AlbertoElDevaluador** [reply]
Really thank you guys!! :3

**Comment by jmmorato** [comment]
> Also you must end the struct definition with a semi colon. It could be that an anonymous array is not supported (don't know all the details of what openddsharp supports), you could try using a typedef as we do here https://github.com/RemedyIT/taox11/blob/d1a33c8563529f907e5ea8a20d04b4ee92dbb0de/tests/array/test.idl#L24

Arrays are supported by the OpenDDSharp IDL generator but, as already pointed, need to be defined via a typedef, that is a requirement from the underlying OpenDDS generator.

**Comment by AlbertoElDevaluador** [comment]
And other question. When i execute the command, all compiled successfull but there arent dataWriter and dataReader... Why?? Why didnt appears and with other IDL yes???

**Reply by jmmorato** [reply]
Did you added the `@topic` annotation to the structure in the IDL definition? 

An example:
https://github.com/jmmorato/openddsharp/blob/f44165dbfe05d01f9cefb59fab75774ae2927e13/Tests/TestIdlJson/IDL/Test.idl#L270C6-L270C6

PS: Another Question, another Discussion. Like that, it will be easier for other users to find the information ;)

**Reply by AlbertoElDevaluador** [reply]
Okey, thank you hehe
